frame/babe/src/lib.rs

```diff
@@ -169,6 +169,22 @@ pub mod pallet {
 		type HandleEquivocation: HandleEquivocation<Self>;
 
 		type WeightInfo: WeightInfo;
+
+		/// Max number of authorities allowed
+		#[pallet::constant]
+		type MaxAuthorities: Get<u32>;
+
+		/// Randomness under construction.
+		///
+		/// We make a tradeoff between storage accesses and list length.
+		/// We store the under-construction randomness in segments of up to
+		/// `MaxSegmentLength`.
+		///
+		/// Once a segment reaches this length, we begin the next one.
+		/// We reset all segments and return to `0` at the beginning of every
+		/// epoch.
+		#[pallet::constant]
+		type MaxSegmentLength: Get<u32>;
```

Comment:
Does it make sense to make it configurable? `UNDER_CONSTRUCTION_SEGMENT_LENGTH` used to be const in this file, but now we're exposing this to runtime constructors?

Reply:
I would ultimately defer to you, but what is the harm? Overhead must be minimal if any?
And it would allow to configure it more flexibly, but I am not familiar enough to make a firm opinion on this.

Reply:
You can make use of `ConstU32` that's defined in `frame_support::traits`.

This const was originally an implementation detail that runtime constructors did not need to worry about. The rationale isn't simply reducing overhead, but preventing leaky abstractions. I would be more open to exposing it to Config if there is an identifiable need for doing so.

Reply:
I have replaced that with a `ConstU32` as suggested, thanks for the tip!